util/src/main/java/io/kubernetes/client/ProtoClient.java

```diff
@@ -69,43 +103,101 @@ public void setApiClient(ApiClient apiClient) {
      * @param path The URL path to call (e.g. /api/v1/namespaces/default/pods/pod-name)
      * @return A Message of type T
      */
-    public <T extends Message> T list(T.Builder listObj, String path) throws ApiException, IOException {
+    public <T extends Message> ObjectOrStatus<T> list(T.Builder listObj, String path) throws ApiException, IOException {
         return get(listObj, path);
     }
 
     /**
+     * Create a Kubernetes API object using protocol buffer encoding. Performs a POST
+     * @param obj The object to create
+     * @param path The URL path to call
+     * @param apiVersion The api version to use
+     * @param kind The kind of the object
+     * @return The response received.
+     */
+    public <T extends Message> ObjectOrStatus<T> create(T obj, String path, String apiVersion, String kind)
+            throws ApiException, IOException {
+        return request(obj.newBuilderForType(), path, "POST", obj, apiVersion, kind);
+    }
+
+    /**
+     * Update a Kubernetes API object using protocol buffer encoding. Performs a PUT
+     * @param obj The object to create
+     * @param path The URL path to call
+     * @param apiVersion The api version to use
+     * @param kind The kind of the object
+     * @return The response received.
+     */
+    public <T extends Message> ObjectOrStatus<T> update(T obj, String path, String apiVersion, String kind)
```

Comment:
is it possible for non-delete calls to return status?

Reply:
Yes, update in particular sometimes returns a 201 (I observed this behavior)